scripts/tools/EXT_Exp_MetricProgrammable.rst

```diff
@@ -0,0 +1,120 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+    t=tags['$t']
+    T=t.upper()
+%>
+:orphan:
+
+.. _ZET_experimental_programmable_metric:
+
+==========================================
+Metric Programmable Experimental Extension
+==========================================
+
+API
+----
+* Enumerations
+
+    * ${t}_metric_programmable_param_type_exp_t
+
+* Structures
+
+    * ${t}_metric_programmable_exp_properties_t
+    * ${t}_metric_programmable_param_exp_desc_t
+    * ${t}_metric_programmable_param_value_exp_t
+    * ${t}_metric_programmable_param_value_exp_desc_t
+    * ${t}_value_exp_desc_t
+    * ${t}_value_uint64_range_exp_t
+
+* Functions
+
+    * ${t}MetricProgrammableGetExp
+    * ${t}MetricProgrammableGetPropertiesExp
+    * ${t}MetricProgrammableGetParamDescriptionExp
+    * ${t}MetricProgrammableGetParamValueDescriptionExp
+    * ${t}MetricCreateFromProgrammableExp
+    * ${t}MetricGroupCreateExp
+    * ${t}MetricGroupAddMetricExp
+    * ${t}MetricDestroyExp
+    * ${t}MetricGroupDestroyExp
+
+Metric Programmable
+~~~~~~~~~~~~~~~~~~~
+
+Application can use ${t}MetricGroupGet to enumerate the list of metric groups and the metrics available for collection.
+However Applications may also want to collect custom metrics which are not listed as part of the enumeration.
+This is made possible by making a list of programmable metrics available to the application, so that it choose different parameter values for the preparing custom metrics.
+The following pseudo-code demonstrates how programmable metrics could be enumerated and custom values could be set to the metric parameters.
+
+Sample Code
+------------
+
+The following pseudo-code demonstrates how programmable metrics could be enumerated and custom values could be set to the metric parameters.
+
+.. parsed-literal::
+
+    ${t}_metric_handle_t * metricHandles;
+    uint32_t metricHandleCount = 0;
+    ${t}_metric_group_handle_t metricGroup;
+    
+    // Query and Get metric programmable handles
+    uint32_t programmableCount = 0;
+    ${t}MetricProgrammableGetExp(device, &programmableCount, nullptr);
+    ${t}_metric_programmable_exp_handle_t * metricProgrammableHandles = allocate(sizeof(${t}_metric_programmable_exp_handle_t) * programmableCount);
+    ${t}MetricProgrammableGetExp(device, &programmableCount, metricProgrammableHandles);
+
+    // Create metrics from metric programmable handles
+    for(uint32_t i = 0; i < programmableCount; i++){
+    
+        ${t}_metric_programmable_exp_handle_t * programmableHandle = metricProgrammableHandles[i];
```

Comment:
is it metric_programmable_exp_handle_t programmableHandle = metricProgrammableHandles[i] here?
(i.e, once you index into the vector, you are getting the handle right?)

Reply:
You are correct.
Will fix

Reply:
fixed